fvm/evm/handler/blockstore.go

```diff
@@ -92,9 +100,56 @@ func (bs *BlockStore) LatestBlock() (*types.Block, error) {
 }
 
 // BlockHash returns the block hash for the last x blocks
-//
-// TODO: implement this properly to keep the last 256 block hashes
-// and connect use it inside the handler to pass as a config to the emulator
-func (bs *BlockStore) BlockHash(height int) (gethCommon.Hash, error) {
-	return gethCommon.Hash{}, nil
+func (bs *BlockStore) BlockHash(height uint64) (gethCommon.Hash, error) {
+	bhl, err := bs.getBlockHashList()
+	if err != nil {
+		return gethCommon.Hash{}, err
+	}
+	if bhl == nil {
```

Comment:
seems to me like this would only happen on uninitialized block hash storage, which I'm not sure whether we should allow, is there a use case for it?

Reply:
I think the case where `getBlockHashList` returns (nil, nil) should not be allowed.

Reply:
Yeah, I am not sure why I implemented it that way in the first place, fixed it at [b9bbee3](https://github.com/onflow/flow-go/pull/5304/commits/b9bbee35aeb17d17160e6a1c95b414b30cf85d3c)